app/decorators/address_form_decorator.rb

```diff
@@ -0,0 +1,42 @@
+class AddressFormDecorator < BaseDecorator
+  def page_title
+    ".page_title.#{i18n_key}"
+  end
+
+  def heading
+    ".heading.#{i18n_key}"
+  end
+
+  def home_address?
+    record.is_a?(HomeAddress)
+  end
+
+  def addresses
+    @addresses ||= begin
+      results = super
+
+      # Add the number of results as the first element of the collection
+      # User has to select something, otherwise there is a validation error
+      results.unshift(address_count_item) if results.any?
+      results
+    end
+  end
+
+  private
+
+  def address_count_item
+    Struct.new(:results_size, :lookup_id) do
+      def compact_address
+        I18n.translate!(:results_count, count: results_size, scope: 'steps.address.results.edit')
+      end
+    end.new(__getobj__.addresses.size)
+  end
+
+  def i18n_key
+    variant.join('.')
+  end
+
+  def variant
+    [record.person.type.underscore, record.type.underscore]
```

Comment:
is the first `.underscore` required here?

Reply:
Yes because `type` is CamelCase, i.e. `Applicant` (or `Partner` when we have it), and we want it all lowercase. For address it can be `HomeAddress` or `CorrespondenceAddress`.

Reply:
Right, understood - just in the case of person it is currently always one word, so it is essentially just to downcase

Reply:
Sorry my comment was not very clear. I intended to say, `underscore` will take the CamelCase and, in addition to lowercase everything, will replace the CamelCase with the underscore version. So for `HomeAddress` it becomes `home_address`, but for just 1 word it still works as intended: `Applicant` becomes `applicant` 😄